.travis.yml

```diff
@@ -1,28 +1,54 @@
-dist: trusty
+dist: xenial
 sudo: false
 language: rust
-cache:
-  directories:
-    - $HOME/.cargo
-    - $HOME/.cache
-    - $TRAVIS_BUILD_DIR/target
-before_cache:
-  - find $TRAVIS_BUILD_DIR/target/debug -maxdepth 1 -type f -delete
+cache: cargo
 
 env:
   global:
     - RUST_BACKTRACE=1
     - RUSTFLAGS="--deny=warnings"
 
-os:
-  - linux
-rust:
-  - stable
-  - beta
-  - nightly
 matrix:
   include:
-  - os: osx
+  - name: linux-stable
+    rust: stable
+    addons:
+      apt:
+        sources:
+          - llvm-toolchain-xenial-7
```

Comment:
Why repeat everything? What's wrong with the original configurations?

Reply:
I found that the three platforms are very different.
For easy understanding and configuration convenience
If don't write into matrix, I have encountered many problems after introducing Windows platform.

Reply:
However, the two Linux platform does not need to be repeated.